File: dotnet/Aspectly.Bridge/src/Protocol/BridgeResultType.cs
using System.Text.Json.Serialization;

namespace Aspectly.Bridge.Protocol;

/// <summary>
/// Result types for bridge responses.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum BridgeResultType
{
    /// <summary>
    /// Method executed successfully.
    /// </summary>
    Success,

    /// <summary>
    /// Method execution failed.
    /// </summary>
    Error
}

File: android/aspectly-bridge/src/main/kotlin/com/aspectly/bridge/protocol/BridgeResultType.kt
package com.aspectly.bridge.protocol

/**
 * Result types for bridge responses.
 * Values match the `@aspectly/core` strings exactly.
 */
enum class BridgeResultType(val value: String) {
    /** Method executed successfully. */
    SUCCESS("Success"),

    /** Method execution failed. */
    ERROR("Error");

    companion object {
        fun fromValue(value: String): BridgeResultType? = entries.firstOrNull { it.value == value }
    }
}
